### PR TITLE
Models documentation

### PR DESCRIPTION
**Pull request by timotheeguerin**

### CODE REVIEW

File: packages/website/src/docs/language-basics/alias.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+id: aliases
+title: Aliases
+---
+
+# Alias
+
+Aliases can be defined for types. This can be helpeful to reuse a complex expression.
+
+Alias is only a syntax helper, it has no represenation in the type graph. This means that aliases cannot be decorated. If wanting to provide a new name look for [`model is`]({{"/docs/language-basics/models" | url}})
```
Comment (bterlson) [inline packages/website/src/docs/language-basics/alias.md:10]:
```suggestion
Alias is only a syntax helper, and it has no representation in the type graph. This means that aliases cannot be decorated. Use [`model is`]({{"/docs/language-basics/models" | url}}) to provide an alternate name for a model.
```

Comment (bterlson) [reply packages/website/src/docs/language-basics/alias.md:10]: Although you may want to mention `@friendlyName` in this context

File: packages/website/src/docs/language-basics/models.md [hunk 2]
```diff
@@ -4,3 +4,150 @@ title: Models
...
+- [Record](#record)
+- [Array](#array)
+
+### Primitives
+
+These are types without any fields(For example `string`, `int32`, `boolean`, etc.)
+
+### Record
+
+Model records are structure with named fields called properties.
```
Comment (bterlson) [inline packages/website/src/docs/language-basics/models.md:24]: I find `model records` to be confusing personally. Record models perhaps? Or just records?

File: packages/website/src/docs/language-basics/alias.md (hunk 1 above)
Comment (timotheeguerin) [reply packages/website/src/docs/language-basics/alias.md]: Well I was really seeing this statement as telling someone that they shouldn't do `alias Foo = string` but `model Foo is string` if they want a new name

File: packages/website/src/docs/language-basics/models.md [hunk 3]
```diff
@@ -4,3 +4,150 @@ title: Models
...
+
+```cadl
+model Dog {
+  address?: string = "wild";
+}
+```
+
+### Array
+
+Array are models behind the scene. The `[]` syntax is just a syntactic sugar for using the `Array` model type.
```
Comment (bterlson) [inline packages/website/src/docs/language-basics/models.md:58]:
```suggestion
Array are models created using the `[]` syntax which is just a syntactic sugar for using the `Array<T>` model type.
```

File: packages/website/src/docs/language-basics/models.md [hunk 4]
```diff
@@ -4,3 +4,150 @@ title: Models
...
+// Dog is equivalent to the following declaration:
+model Dog {
+  species: string;
+  name: string;
+}
+```
+
+### Extends
+
+Sometimes you want to create an explicit relationship between two models, for example when you want to emit class definitions in languages which support inheritance. The `extends` keyword can be used to establish such a relationship. It is also used extensively with `interface` to compose from existing interface building blocks.
```
Comment (bterlson) [inline packages/website/src/docs/language-basics/models.md:89]:
```suggestion
Sometimes you want to create an explicit relationship between two models, for example when you want to emit class definitions in languages which support inheritance. The `extends` keyword can be used to establish such a relationship. 
```

This seems out of scope in this section of the doc.

File: packages/website/src/docs/language-basics/models.md [hunk 5]
```diff
@@ -4,3 +4,150 @@ title: Models
...
+model Animal {
+  species: string;
+}
+
+model Dog extends Animal {}
+```
+
+## Is
+
+Sometimes you want to copy all aspects of a type without creating a nominal inheritance relationship. The `is` keyword can be used for this purpose. It is like spread, but also copies [decorators]({{"/docs/language-basics/decorators" | url}}) in addition to properties. One common use case is to give a better name to a [template](#Templates) instantiation:
```
Comment (bterlson) [inline packages/website/src/docs/language-basics/models.md:101]: I feel like a better explanation is "Sometimes you want to create a new type that is an exact copy of an existing type but with some additional properties or metadata"?

Comment (bterlson) [reply packages/website/src/docs/language-basics/models.md:101]: "Like spread" seems confusing, you can't do multiple "is". You could perhaps say "it copies all of the properties (like spread)".

**Review by bterlson — APPROVED** [review]
